### ISSUE TITLE
Table relationships showing up as hidden in query builder when not

### ISSUE THREAD

**Issue by tlammer** [specify/specify7#1845]
Discovered by Willem

Preparation property is showing up as hidden in the query builder but neither the field nor the table are hidden.  I unhid the field in the schema (preparation.preparationproperty), confirmed the table itself is not hidden, then created a new query.  Preparation property is listed under hidden relationships

**Comment by maxxxxxdlp** [comment]
When we were working on the workbench mapper, we found all kinds of relationships that are visible in the default schema configuration, but probably should not be exposed in the workbench mapper.

There were so many of these "exceptions", that I had to develop an entire system that would reclassify some hardcoded list of fields and relationships as hidden, optional, readonly, required, system or even remove them completely (even from the `Reveal Hidden Fields`).

The full list of such overrides is here:
https://github.com/specify/specify7/blob/44fe80fc9a9d8da3112af97afaad8487811fcafd/specifyweb/frontend/js_src/lib/schemaoverrides.ts#L64-L201

I would be strongly in favor of getting rid of most if not all of those overrides. As it currently stands, it is impossible for people to unhide those fields, even if they are marked as unhidden in the schema config.
